releasenotes-builder/src/main/java/com/github/checkstyle/Main.java

```diff
@@ -66,6 +70,26 @@ public static void main(String... args) {
                 final Result notesBuilderResult = runGithubNotesBuilder(cliOptions);
                 errorCounter = notesBuilderResult.getErrorMessages().size();
                 if (errorCounter == 0) {
+                    final char[] charArray = cliOptions.getReleaseNumber().toCharArray();
```

Comment:
move you code out of this IF, and if version mismatch happening, errrorCunter++.
and do not do any publishing.

Reply:
Changes Done